packages/jest-circus/src/legacy_code_todo_rewrite/jest-adapter-init.js

```diff
@@ -157,11 +161,18 @@ const eventHandler = (event: Event) => {
     case 'test_success':
     case 'test_failure': {
       _addSuppressedErrors(event.test);
+      _addExpectedAssertionErrors(event.test);
       break;
     }
   }
 };
 
+const _addExpectedAssertionErrors = (test: TestEntry) => {
+  const errors = extractExpectedAssertionsErrors();
+  errors.length && (test.status = 'fail');
```

Comment:
i would disagree with readability, recently i find it very hard to navigate through 100 lines functions that do just 2 things, but i agree that this line messes up semantics pretty bad by using a logical operator as an `if` condition :)